### DISCUSSION TITLE
Button with resizable image/icon

### DISCUSSION THREAD

**Discussion by aiuliansoft** [dotnet/maui#21381] [Ideas]
Hi all,

I was struggling the other days with some buttons having different sizes. The main idea is that those buttons should contain a Text and an Image/Icon. 

My problem was that I had to use the same svc icon for multiple sizes of the button, but the size of the Icon remained unchanged and apparently there's no support in MAUI to resize that icon inside the button.

Btw, it's about <Button>, bot <ButtonImage>, cause this one the latest, has not support for a Text.

I'm pretty new in MAUI development, so don't know much about it, but as far as I read in the official documentation and other comunity threads, there's not support yet.

Have anybody encounted this and managed to overcome it?

Thanks,

**Comment by prabhavmehra** [comment]
One of the things we did was creating a markup extension and added support for additional properties like color, size, name. They are basically FontImageSource so very easy to configure using FontSize (as you would in labels).

**Reply by aiuliansoft** [reply]
Hi,

Thank you for your answer. Could you please share a code snippet from your solution? I'm new to MAUI so would be very helpful. 

Regards,
Iulian

**Reply by prabhav-mehra** [reply]
I don't think I can provide a code snippet for now. 

But this is how we did it,

Implement a markup extension which has an aim to provide a FontImageSource. 

Create bindable properties for color, size and glyph documentation here https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.fontimagesource?view=net-maui-8.0

Now using these properties create a FontImageSource and return it. 

Now just call this in your xaml for your button

ImageSource="{local:iconextension glyph="...", color="...",size="..."}"

**Reply by aiuliansoft** [reply]
Alright. Thank you for your answer!
